src/rtc.c

```diff
@@ -0,0 +1,518 @@
+        0xffffffffffffffff,
+        0xfffe000000000000,
+        0,
+        0,
+        0,
+        0,
+        1
+    };
+    static uint64_t output[8];
+
+    input[0] = (input[0] & 0xFFFFFFFFFFFFFF00) | block;
+
+    controller_exec_pif( input, output );
+
+    *data = output[1];
+
+    return output[2] >> 56;
+}
+
+/**
+ * @brief Write a block of data to the real-time clock
+ *
+ * @param[in]   block
+ *              Which RTC block to write to (0-2)
+ *
+ * @param[out]  data
+ *              RTC block data to write
+ *
+ * @return the busy status of the real-time clock
+ */
+uint8_t joybus_rtc_write( uint8_t block, const uint64_t * data )
+{
+    assert(block <= 2);
+
+    static uint64_t input[8] =
+    {
+        0x000000000A010800,
+        0xffffffffffffffff,
+        0xfffe000000000000,
+        0,
+        0,
+        0,
+        0,
+        1
+    };
+    static uint64_t output[8];
+
+    input[0] = (input[0] & 0xFFFFFFFFFFFFFF00) | block;
+    input[1] = *data;
+
+    controller_exec_pif( input, output );
+
+    return output[2] >> 56;
+}
+
+/**
+ * @brief Read the status of the real-time clock.
+ *
+ * Some flash carts require the RTC to be explicitly enabled.
+ * Some emulators and flash carts do not support RTC at all.
+ *
+ * @return whether the Joybus RTC is present on the cartridge.
+ */
+bool joybus_rtc_is_present( void )
+{
+    return (joybus_rtc_status() >> 8) == JOYBUS_RTC_IDENTIFIER;
+}
+
+/**
+ * @brief Read the status of the real-time clock.
+ *
+ * The RTC will be busy when the control data is in
+ * #JOYBUS_RTC_CONTROL_MODE_SET and is ready to be written to.
+ *
+ * @return whether the Joybus RTC status is busy
+ */
+bool joybus_rtc_is_busy( void )
+{
+    return (joybus_rtc_status() & 0xFF) == JOYBUS_RTC_STATUS_BUSY;
+}
+
+/**
+ * @brief Read the control block from the real-time clock.
+ *
+ * The control data should generally be one of the pre-defined modes:
+ * #JOYBUS_RTC_CONTROL_MODE_SET when writing to the RTC.
+ * #JOYBUS_RTC_CONTROL_MODE_RUN when not writing to the RTC.
+ *
+ * The calibration data is an opaque value that should be propagated when
+ * calling #joybus_rtc_write_control. Emulators and flash carts may return
+ * zeroes for this data, which is also fine. Based on reverse-engineering
+ * the real hardware, this data is simply described as "magical", but
+ * should be preserved in the interest of high-accuracy and compatibility.
+ *
+ * Generally, you should not need to call this function directly. Prefer
+ * calling the high-level #joybus_rtc_run and #joybus_rtc_set functions which
+ * handle the delays and calibration propagation. Call #joybus_rtc_status or
+ * #joybus_rtc_is_busy to determine which control mode the RTC is in.
+ *
+ * @param[out]  control
+ *              Destination pointer for the RTC control data.
+ *
+ * @param[out]  calibration
+ *              Destination pointer for the RTC calibration data.
+ */
+void joybus_rtc_read_control( uint16_t * control, uint32_t * calibration )
```

Comment:
Static

Reply:
Addressed in https://github.com/DragonMinded/libdragon/pull/152/commits/e46fbb07a6cbb01a98f05a2d3b170d4ff1860b77